test/Extensions/InputSystem/DeviceInputBehavior.test.ts

```diff
@@ -0,0 +1,159 @@
+import { Mouse } from "../../../src/Extensions/InputSystem/Mouse";
+import { describe, it, expect, vi, beforeAll, beforeEach } from "vitest";
+import { GameEngineWindow } from "../../../src/Core/GameEngineWindow";
+import { InputGameEngineComponent } from "../../../src/Extensions/InputSystem/InputGameEngineComponent";
+import { InputUtility } from "./InputUtility";
+import { Keyboard } from "../../../src/Extensions/InputSystem/Keyboard";
+import { MockDeviceInputBehavior } from "./Mocks/MockDeviceInputBehavior";
+
+describe("DeviceInputBehavior", (): void => {
+  let mouse: Mouse;
+  let deviceInputBehavior: MockDeviceInputBehavior;
+  let inputGameEngineComponent: InputGameEngineComponent;
+  let gameEngineWindow: GameEngineWindow;
+  let keyboard: Keyboard;
+
+  beforeAll((): void => {
+    InputUtility.mockDocumentEventListeners();
+    keyboard = new Keyboard();
+    mouse = new Mouse();
+
+    inputGameEngineComponent = new InputGameEngineComponent();
+    gameEngineWindow = GameEngineWindow.instance;
+
+    inputGameEngineComponent.addDevice(keyboard);
+    inputGameEngineComponent.addDevice(mouse);
+    gameEngineWindow.addGameComponent(inputGameEngineComponent);
+  });
+
+  beforeEach((): void => {
+    deviceInputBehavior = new MockDeviceInputBehavior();
+  });
+
+  it("should change the value of function called to onMouseLeftClickUp", (): void => {
+    // Given
+    deviceInputBehavior.onEnable();
```

Comment:
You'd probably want to attach this to a gameobject instead of calling onEnable manually....